marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java

```diff
@@ -709,6 +709,7 @@ private String generateAndEncodeURI(String path) throws  URISyntaxException {
 
     private void applyPermissions(String permissions, DocumentMetadataHandle metadataHandle) {
         String[] perms = permissions.split(",");
+        Map<String, List<DocumentMetadataHandle.Capability>> permMap = new HashMap<>();
```

Comment:
@srinathgit I just remembered that there's a class in ml-javaclient-util that does this already - DefaultDocumentPermissionsParser.  It's being used in DHF already. You give it the permissions string and the DocumentPermissions object from DocumentMetadataHandle, and it handles the rest. I recommend using that.

Reply:
Yes, please reuse this method, it keeps us static in how we handle the permissions string across the datahub.